src/capi/impl/capi_sframe.cpp

```diff
@@ -169,21 +195,24 @@ EXPORT tc_sframe* tc_sframe_join_on_single_column(tc_sframe* left,
 
   CHECK_NOT_NULL(error, left, "left tc_sframe", NULL);
   CHECK_NOT_NULL(error, right, "right tc_sframe", NULL);
-
-  return new_tc_sframe(
-      left->value.join(right->value, {std::string(column)}, how));
+  CHECK_NOT_NULL(error, column, "column", NULL);
+  CHECK_NOT_NULL(error, how, "how", NULL);
+
+  tc_flex_list *join_columns = tc_flex_list_create(error);
+  if (join_columns == NULL) { assert(false); return NULL; } // error should be populated already
+  tc_flexible_type *flex_column = tc_ft_create_from_cstring(column, error);
+  if (flex_column == NULL) { assert(false); return NULL; } // error should be populated already
```

Comment:
In theory, an error here will leak `join_columns`. But the likelihood is quite low. We should have a cleanup issue to simplify writing code here.

Since we can use C++ in the implementation of this C API, probably we should define some aliases for e.g. std::unique_ptr<tc_flex_list, /* appropriate deleter */>